Document TRL and DSOS (#845)
corona-warn-app/cwa-documentation: solution_architecture.md

For Android devices, the features are integrated into the [Google Play Services](https://developers.google.com/android/exposure-notifications/exposure-notifications-api#architecture), which means that only this specific application needs to be updated for it to work. Devices starting with Android 6.0 (API version 23) and integrated BLE chips are [supported](https://developers.google.com/android/exposure-notifications/exposure-notifications-api#architecture).

### ENF Usage

> The Exposure Notification Framework (ENF) encapsulates handling of the keys, including all cryptographic operations on them, and broadcasting and scanning for Bluetooth advertisement packages for the app (see *Figure 11* and *Figure 12*). The app itself does not have access to the collected exposures, i.e. the Rolling Proximity Identifiers, and neither is it informed if a new one has been collected by the framework. The only output of the ENF upon an infection is a collection of temporary exposure keys as shown in *Figure 11*. Those are subsequently called diagnosis keys.

Reviewer: Suggestion: Add the information that the app also is not informed if these keys are deleted. This is quite an important info and answers questions like: "Why doesn't the app change if I delete the keys via OS settings?"

Author: Fixed with https://github.com/corona-warn-app/cwa-documentation/pull/845/commits/0e19cdfc1325f4af10322efb5237cc471cef2750.

Each exposure window contains the following information:

- **infectiousness** and **report type** - these parameters are attached to the respective diagnosis key by the sending app.
- **day of the exposure** - this parameter is determined by ENF based on when the respective RPI was received. Note that even though precise timestamp information exists in ENF, only the day itself is exposed (i.e. YYYY-MM-DD).
- **multiple scan instances** - this parameter represents occurrences where the other device has been actively identified during the scanning process. A scan instance consists of "seconds since last scan", i.e. for how long the other device was identified, and attenuation information as a measure of distance between the devices.

### Risk Calculation

> The Exposure Windows form the foundation of the risk calculation in the app. The result is an easy-to-understand risk level that is displayed to the user: low risk (i.e. _green_) or high risk _red_. The details of the risk calculation, e.g. exposure windows or intermediate results, are not visible to the user. Further information regarding the individual exposure events (such as the matched Rolling Proximity Identifier, the Temporary Exposure Key or the exact time) remains within the secure storage of the framework and cannot be retrieved by the application.

Reviewer: Add brackets to the word "red".

```suggestion
The Exposure Windows form the foundation of the risk calculation in the app. The result is an easy-to-understand risk level that is displayed to the user: low risk (i.e. _green_) or high risk (_red_). The details of the risk calculation, e.g. exposure windows or intermediate results, are not visible to the user. Further information regarding the individual exposure events (such as the matched Rolling Proximity Identifier, the Temporary Exposure Key or the exact time) remains within the secure storage of the framework and cannot be retrieved by the application.
```

### Days Since Onset of Symptoms

The "days since onset of symptoms" (DSOS) is a relative numeric measure that indicates the infectiousness of a diagnosis key. The reference point is the date of onset of symptoms that a user can provide when uploading diagnosis keys.

It is optional to provide information about the symptom onset and there different levels of granularity. This results in five different patterns for determining the DSOS of a diagnosis key and numeric values between -14 and 4000.

The different patterns are:

1. **No information** - the user declines to provide information about symptom onset

    <details>
    <summary>Figure 17: Determine DSOS for 'No information'</summary>

    | ![Figure 17: Determine DSOS for 'No information'](images/solution_architecture/dsos-no-information.png "Figure 17: Determine DSOS for 'No information'") |
    |:--:|
    | **Figure 17: Determine DSOS for 'No information'**|

    </details>

> 2. **No symptoms** - the user indicates that he/she has symptoms

Reviewer 2: "the user indicates that he/she has symptoms" =>

"the user indicates that he/she has no symptoms" ?

Author: Thanks, good catch (as usual) 👍 Fixed with https://github.com/corona-warn-app/cwa-documentation/pull/845/commits/b1505f1cb5a031acd2586d89ec567282a2649c23.

> 2. **No symptoms** - the user indicates that he/she has no symptoms

Reviewer: Use they to be gender neutral.

```suggestion
2. **No symptoms** - the user indicates that they have no symptoms
```

Author: @Ein-Tim I like the proposal of dropping _he/she_ but I'm not sure if _they_ is such a good replacement. How about just dropping the relative clause and rephrase it to _the user has no symptoms_?

Reviewer: Sounds good!

Author: fixed with https://github.com/corona-warn-app/cwa-documentation/pull/845/commits/19341c3fef9edf172d14c7773f933263b71ddefc

    <details>
    <summary>Figure 18: Determine DSOS for 'No symptoms'</summary>

    | ![Figure 18: Determine DSOS for 'No symptoms'](images/solution_architecture/dsos-no-symptoms.png "Figure 18: Determine DSOS for 'No symptoms'") |
    |:--:|
    | **Figure 18: Determine DSOS for 'No symptoms'**|

    </details>

> 3. **No date information** - the user indicates that he/she has symptoms but declines to provide any date information

Reviewer: ```suggestion
3. **No date information** - the user indicates that they have symptoms but decline to provide any date information
```

    <details>
    <summary>Figure 19: Determine DSOS for 'No date information'</summary>

    | ![Figure 19: Determine DSOS for 'No date information'](images/solution_architecture/dsos-date-unknown.png "Figure 19: Determine DSOS for 'No date information'") |
    |:--:|
    | **Figure 19: Determine DSOS for 'No date information'**|

    </details>

> 4. **Date range** - the user indicates that he/she has symptoms and provides a date range (e.g. "within the last week", "1 to 2 weeks ago", "more than 2 weeks ago")

Reviewer: ```suggestion
4. **Date range** - the user indicates that they have symptoms and provide a date range (e.g. "within the last week", "1 to 2 weeks ago", "more than 2 weeks ago")
```

    <details>
    <summary>Figure 20: Determine DSOS for 'Date range'</summary>

    | ![Figure 20: Determine DSOS for 'Date range'](images/solution_architecture/dsos-date-range.png "Figure 20: Determine DSOS for 'Date range'") |
    |:--:|
    | **Figure 20: Determine DSOS for 'Date range'**|

    </details>

> 5. **Specific date** - the user indicates that he/she has symptoms and provides a specific date

Reviewer: ```suggestion
5. **Specific date** - the user indicates that they have symptoms and provide a specific date
```
